### ISSUE TITLE
BUG: Fatal error with wrong --archive type

### ISSUE THREAD

**Issue by lbb** [GoogleContainerTools/krew#109]
When using `--archive` we assume the filetype of the index URL.
This will lead to a decoding error when the `--archive` type mismatches with the type specified in the `uri:`.

```text
./bin/krew-darwin_amd64 install --archive ./krew.zip --manifest krew.yaml -v 6
I1014 21:40:35.421829    9181 install.go:132] --manifest specified, not ensuring plugin index
I1014 21:40:35.423021    9181 install.go:101] Will install plugin: krew
Installing plugin: krew
I1014 21:40:35.423051    9181 install.go:74] Looking for installed versions
I1014 21:40:35.423066    9181 util.go:77] Searching for installed versions of krew in "/Users/lbb/.krew/bin"
I1014 21:40:35.423113    9181 install.go:83] Finding download target for plugin krew
I1014 21:40:35.423123    9181 util.go:36] Using os=darwin arch=amd64
I1014 21:40:35.423129    9181 util.go:59] Matching platform for labels(arch=amd64,os=darwin)
I1014 21:40:35.423179    9181 util.go:66] Found matching platform with index (0)
I1014 21:40:35.423186    9181 util.go:129] Matching plugin version is 170f15a0f363bf1d35896b5963495e70e215caa41ebfb7244351c42c6ab06222
I1014 21:40:35.423196    9181 install.go:46] Creating download dir "/var/folders/b6/3hk_x4nx08z2qb9ph8_zcxgh0000gn/T/krew-downloads/krew"
I1014 21:40:35.423865    9181 install.go:61] Getting sha256 (170f15a0f363bf1d35896b5963495e70e215caa41ebfb7244351c42c6ab06222) signed version
I1014 21:40:35.423888    9181 downloader.go:36] Fetching "https://github.com/GoogleContainerTools/krew/releases/download/v0.1.0-alpha.1-56-g1727d93/krew.tar.gz"
I1014 21:40:35.423936    9181 downloader.go:43] Reading download data into memory
I1014 21:40:35.507490    9181 downloader.go:48] Read 14405237 bytes of download data into memory
W1014 21:40:35.507655    9181 install.go:114] failed to install plugin "krew"
F1014 21:40:35.507691    9181 root.go:50] failed to install some plugins: [krew]
github.com/GoogleContainerTools/krew/cmd/krew/cmd.init.1.func1
	/Users/lbb/go/src/github.com/GoogleContainerTools/krew/cmd/krew/cmd/install.go:124
github.com/GoogleContainerTools/krew/vendor/github.com/spf13/cobra.(*Command).execute
	/Users/lbb/go/src/github.com/GoogleContainerTools/krew/vendor/github.com/spf13/cobra/command.go:762
github.com/GoogleContainerTools/krew/vendor/github.com/spf13/cobra.(*Command).ExecuteC
	/Users/lbb/go/src/github.com/GoogleContainerTools/krew/vendor/github.com/spf13/cobra/command.go:852
github.com/GoogleContainerTools/krew/vendor/github.com/spf13/cobra.(*Command).Execute
	/Users/lbb/go/src/github.com/GoogleContainerTools/krew/vendor/github.com/spf13/cobra/command.go:800
github.com/GoogleContainerTools/krew/cmd/krew/cmd.Execute
	/Users/lbb/go/src/github.com/GoogleContainerTools/krew/cmd/krew/cmd/root.go:48
main.main
	/Users/lbb/go/src/github.com/GoogleContainerTools/krew/cmd/krew/main.go:23
runtime.main
	/Users/lbb/homebrew/Cellar/go/1.11.1/libexec/src/runtime/proc.go:201
runtime.goexit
	/Users/lbb/homebrew/Cellar/go/1.11.1/libexec/src/runtime/asm_amd64.s:1333
```

**Comment by lbb** [comment]
It also seems that we still download the archive

**Comment by lbb** [comment]
The sha256 does not match, but it gets not shown.
I'll add logging.

**Comment by ahmetb** [comment]
This is fixed in #103.
